airflow/serialization/serialized_objects.py

```diff
@@ -53,14 +55,23 @@
 log = logging.getLogger(__name__)
 FAILED = 'serialization_failed'
 
-BUILTIN_OPERATOR_EXTRA_LINKS: List[str] = [
-    "airflow.providers.google.cloud.operators.bigquery.BigQueryConsoleLink",
-    "airflow.providers.google.cloud.operators.bigquery.BigQueryConsoleIndexableLink",
-    "airflow.providers.google.cloud.operators.mlengine.AIPlatformConsoleLink",
-    "airflow.providers.qubole.operators.qubole.QDSLink",
+BUILTIN_OPERATOR_EXTRA_LINKS: Set[str] = {
     "airflow.operators.dagrun_operator.TriggerDagRunLink",
     "airflow.sensors.external_task_sensor.ExternalTaskSensorLink",
-]
+}
+
+cache = lru_cache(maxsize=None)
+
+
+@cache
```

Comment:
Shouldn't we cache ProviderManager instances? It seems to me that creating this object is the most expensive, but not reading the extra_links_class_names attribute?

Reply:
This cache is to be removed. I've changed ProviderManager to be singleton so it does not need caching either.

Reply:
Ah no. It is ok here. I want to still get it from the cache so i do not update the dict all the time only the first time. it's really small optimisation, but I just wanted to make sure it is only set once. Otherwise I'd have to re-create the dictionary every time (and as mentioned above ProvidersManager is a singleton, so no need to cache anything there.).